### PR TITLE
Allow MemGPT to read/write text files + make HTTP requests

### PR DESCRIPTION
**Pull request by cpacker**
Addresses #26 

---

TODO

- [x] Allow toggling of preset from CLI (right now it's hardcoded)

---

## Example with MemGPT launching HTTP requests

image

---

## Example with MemGPT reading / writing from a shared .txt file

image

```sh
$ cat notes.txt
I will tell you a secret
```

Modify the file, then check if MemGPT can read the updated file:

```sh
$ echo "I like dogs" >> notes.txt
$ cat notes.txt
I will tell you a secret
I like dogs
```

MemGPT can successfully read from the updated file (after some prodding to read more lines):

image

---

### CODE REVIEW

**Comment by cpacker** [comment]
@vivi do we have to do anything special to support toggling the preset in the CLI?

Or can we simply add another arg here (and up the stack): https://github.com/cpacker/MemGPT/blob/main/memgpt/main.py#L165

File: memgpt/main.py [hunk 1]
```diff
@@ -304,7 +304,8 @@ async def main(
     chosen_persona = cfg.memgpt_persona
 
     memgpt_agent = presets.use_preset(
-        presets.DEFAULT,
+        # presets.DEFAULT,
+        "memgpt_extras",
```
Comment (cpacker) [inline memgpt/main.py:308]: Need to rebase on main and test that we can launch the "extras" preset using the preset flag/config.

File: memgpt/main.py [hunk 2]
```diff
@@ -445,15 +445,15 @@ async def run_agent_loop(memgpt_agent, first, no_verify=False, cfg=None, legacy=
                     continue
 
                 elif user_input.lower() == "/dump":
-                    await print_messages(memgpt_agent.messages)
+                    await memgpt.interface.print_messages(memgpt_agent.messages)
```
Comment (cpacker) [inline memgpt/main.py:448]: This patches a bug

**Comment by cpacker** [comment]
Still need to add preset selection, put it separate PR

**Review by vivi — APPROVED** [review]
😈 LGTM

**Pull request merged**